Makefile

```diff
@@ -190,27 +190,26 @@ unit-sup: build-launcher-for-supervisor-tests
 .PHONY: build-launcher-for-supervisor-tests
 
 # Lints we need to work through and decide as a team whether to allow or fix
-UNEXAMINED_LINTS = clippy::cyclomatic_complexity \
-                   clippy::large_enum_variant \
-                   clippy::len_without_is_empty \
-                   clippy::module_inception \
-                   clippy::needless_pass_by_value \
-                   clippy::needless_return \
-                   clippy::new_ret_no_self \
-                   clippy::new_without_default \
-                   clippy::new_without_default_derive \
-                   clippy::question_mark \
-                   clippy::redundant_field_names \
-                   clippy::too_many_arguments \
-                   clippy::trivially_copy_pass_by_ref \
-                   clippy::wrong_self_convention \
-                   renamed_and_removed_lints
+UNEXAMINED_LINTS =
 
 # Lints we disagree with and choose to keep in our code with no warning
-ALLOWED_LINTS =
+ALLOWED_LINTS = clippy::module_inception \
+                clippy::needless_return \
+                clippy::new_ret_no_self \
+                clippy::new_without_default \
+                clippy::new_without_default_derive \
+                clippy::redundant_field_names
 
 # Known failing lints we want to receive warnings for, but not fail the build
-LINTS_TO_FIX =
+LINTS_TO_FIX = clippy::cyclomatic_complexity \
+               clippy::large_enum_variant \
+               clippy::len_without_is_empty \
+               clippy::needless_pass_by_value \
```

Comment:
https://rust-lang-nursery.github.io/rust-clippy/master/index.html#needless_pass_by_value

The argument in favor of fixing is that we stand to eliminate unnecessary overhead. The argument against is that we do this in *a lot* of places. Possibly we should try fixing some see if a stronger feeling develops.

[needless_pass_by_value.txt](https://github.com/habitat-sh/habitat/files/2814059/needless_pass_by_value.txt)

Reply:
I like this one. There might be a lot of warnings initially, but I feel like it'll be good in the long run.

Reply:
This is definitely something we should fix, and I'm guessing it will be fairly instructive as well.  I know when I first started with Rust, I passed everything by value by default, just because that's more common in other languages.

Reply:
+1.  We should do this.